instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -204,10 +209,11 @@ def _trace_postrun(self, *args, **kwargs):
             span.set_attribute(_TASK_NAME_KEY, task.name)
 
         activation.__exit__(None, None, None)
-        utils.detach_span(task, task_id)
+        utils.detach_context(task, task_id)
         self.update_task_duration_time(task_id)
         labels = {"task": task.name, "worker": task.request.hostname}
         self._record_histograms(task_id, labels)
+        context_api.detach(token)
```

Comment:
It seems this line introduces a bug where the token could be "None" if the prerun signal has no traceparent.

This could happen for example if the process sending the task isn't instrumented, while the celery worker is instrumented. This results in an error since token cannot be None when calling context_api.detach()

This seems related as well: https://github.com/open-telemetry/opentelemetry-python/issues/4163

Reply:
Good catch!

Reply:
This was a silly mistake on my end. I'll get a fix for this tonight.

Reply:
started a draft PR here: https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2855

still trying to figure out how to write a test for this

Reply:
If an integration test is hard to write maybe an unit test could help in moving things forward